src/control/lib/netdetect/netdetect.go

```diff
@@ -73,91 +76,102 @@ func initLib(xmlTopology string) (C.hwloc_topology_t, error) {
 
 // cleanUp closes out the hwloc resources
 func cleanUp(topology C.hwloc_topology_t) {
-	if (topology != nil) {
+	if topology != nil {
 		C.hwloc_topology_destroy(topology)
 	}
 }
 
-// GetAffinityForNetworkDevices searches the system topology for each network
-// device specified by the input string and returns corresponding cpuset and
-// nodeset bitmaps as strings.  When converted back to hwloc_bitmap_t via
+// GetAffinityForNetworkDevices searches the system topology reported by hwloc
+// for the devices specified by deviceNames and returns the corresponding
+// name, cpuset and nodeset information for each device it finds.
+//
+// The input deviceNames []string specifies names of each network device to
+// search for. Typical network device names are "eth0", "eth1", "ib0", etc.
+//
+// The DeviceAffinity struct specifies a name, cpuset and nodeset strings.
+// The DeviceAffinity DeviceName matches the deviceNames strings and should be
+// used to help match an input device with the output.
+// The DeviceAffinity CPUSet and NodeSet are string representations of the
+// corresponding hwloc bitmaps.  When converted back to hwloc_bitmap_t via
 // hwloc_bitmap_sscanf() these bitmaps are used by the hwloc API to bind a
 // thread to processing units that are closest to the given network device.
 //
-// The input netNames string has the form "device0,device1,...deviceN"
-// where 'device' specifies a network device such as "eth0", "eth1", or "ib0"
-// The return string has the form:
-// "dev0:cpuset:nodeset;dev1:cpuset:nodeset; ... devN:cpuset:nodeset"
-// where dev is the name of the network device and cpuset and nodeset are
-// the corresponding bitmap strings.
-//
 // Network device names that are not found in the topology are ignored.
 // The order of network devices in the return string depends on the natural
 // order in the system topology and does not depend on the order specified
 // by the input string.
-func GetAffinityForNetworkDevices(netNames string,
-	xmlTopology string) (string, error) {
-
-	var affinity string
+func GetAffinityForNetworkDevices(deviceNames []string) (*[]DeviceAffinity, error) {
```

Comment:
*[]DeviceAffinity is very different from []*DeviceAffinity. In go, you almost never want to have a pointer to a slice.

Reply:
Ok, I will look at it and fix.

Reply:
To be clear: I think for the sake of simplicity, you shouldn't worry about pointers to the DeviceAffinity structs either. Just returning []DeviceAffinity should be fine.

Reply:
That helps.  Thank you.